Source/EasyNetQ/PersistentConnection.cs

```diff
@@ -100,6 +102,30 @@ public void Dispose()
             }
         }
 
+        private IModel GetModel(IAutorecoveringConnection connection)
+        {
+            if (connection.ChannelMax == 0) {
+                // unlimited channels available: no need to manage channels
+                return connection.CreateModel();
+            }
+
+            lock (mutex)
+            {                   
+                if (models.Count < connection.ChannelMax)
+                {
+                    var model = connection.CreateModel();
+                    model.ModelShutdown += (sender, args) => models.Remove(model);
```

Comment:
It seems to be not thread safe: one thread calls `models.Add(model)`, another one sends model.ModelShutdown event and `models.Remove(model)`, as a result we have two thread modifying `List`.

Reply:
👍